server/src/test/java/io/crate/integrationtests/DDLIntegrationTest.java

```diff
@@ -978,4 +978,38 @@ public void test_geo_shape_can_be_not_null() {
         ));
 
     }
+
+    @Test
+    public void test_add_geo_shape_column_keeps_geo_settings_and_adds_generated_expression() throws Exception {
+        execute("CREATE TABLE tbl (id INTEGER)");
+
+        execute("ALTER TABLE tbl ADD COLUMN g geo_shape " +
+            "GENERATED ALWAYS AS 'POLYGON (( 5 5, 30 5, 30 30, 5 30, 5 5 ))' " +
+            "INDEX using QUADTREE with (precision='123m', distance_error_pct='0.123')"
```

Comment:
Looks like this is not related to `ADD COLUMN` but also happen if the column was defined on table creation. Thus I'd suggest to use a test which defines it on table creation (and rename the test according to that) to avoid any confusion. Otherwise one could think that this is dedicated to the alter statement.

Reply:
Yes, it happens on creation as well. I covered it using `ADD COLUMN` since it was a bug very specific to the new [PR](https://github.com/crate/crate/pull/13238/files) - so even if I fix it here, CREATE would pass but ADD COLUMN would still fail on the ADD COLUMN implementation and idea was to have only 1 test for this issue.

Would it be fine if I use `CREATE` in this PR but add similar test for `ADD COLUMN in another PR?

Reply:
Ah ok, thanks for this clarification. What about adding a `DocIndexMetadataTest` and `MetadataToASTNodeResolverTest` test case to validate this fix? Would this also validate the `ADD COLUM` scenario?

Reply:
Updated tests in https://github.com/crate/crate/pull/13254/commits/85c9b8a09a3e3e46e087c43bbd83a9fc19a7cdf0

`ADD COLUMN` would need separate test case as it's basically same bug (forgetitng to extract actual Reference out of GeneratedReference) but in the new ADD COLUMN transport action. But as it's a different scenario I think it would be fine to cover it with another test.